File: Java/LinkedList.java
class Node {
    int data;
    Node next;

    Node(int data){
        this.data = data;
        this.next = null;
    }
}

class LinkedList{
    Node head = null;
    
    void InsertBegging(int data){
        Node newNode = new Node(data);
        newNode.next = head;
        head = newNode;
    }

    void InsertLast(int data){
        if(head == null){
            InsertBegging(data);
        }
        Node newNode = new Node(data);
        Node temp = head;
        while(temp.next != null){
            temp = temp.next;
        }
        temp.next = newNode;
    }

    void InsertIndex(int index, int data){
        if(index < 0){
            System.out.println("Invalid Syntax");
            return;
        }

        Node temp = head;
        for(int i =0; i<index-1; i++){
            temp = temp.next;
        }
        Node newNode = new Node(data);
        newNode.next = temp.next;
        temp.next = newNode;
    }

    void Delete(int data){
       Node temp = head;

       if(temp.data == data){
        temp = temp.next;
        return;
       }

       while(temp.next != null && temp.next.data != data){
        temp = temp.next;
       }

       if(temp.next == null){
        System.out.println("Invalid Index");
        return;
       }
       
       temp.next = temp.next.next;

    }

    void DeleteIndex(int index){
        Node temp = head;
        if(index == 0){
            System.out.println("Invalid Index");
            return;
        }
        if(index == 0){
            temp = temp.next;
            return;
        }

        for(int i = 0; i<index-1; i++){
            temp = temp.next;
        }
        temp.next = temp.next.next;
    }

    void Search(int data){
        Node temp = head;
        int count = 0;
        while(temp != null){
            if(temp.data == data){
                System.out.println("Data found at Index " + count);
                return;
            }
            temp = temp.next;
            count++;
        }
        System.out.println("Data not found");
    }

    void display(){
        Node temp = head;
        while(temp != null){
            System.out.print(temp.data + " -> ");
            temp = temp.next;
        }
        System.out.print("Null");
    }

    public static void main(String args[]){
        LinkedList list = new LinkedList();
        
        list.InsertBegging(1);
        list.InsertBegging(2);
        list.InsertBegging(3);
        list.InsertLast(4);
        list.InsertIndex(2, 5);
        list.Delete(5);
        list.DeleteIndex(1);
        list.Search(5);

        list.display();
    }
}

File: Python/LinkedList/LinkedList.py

class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def InsertBegining(self,data):
        new_Node = Node(data)
        new_Node.next = self.head
        self.head = new_Node

    def InsertEnd(self,data):
        new_Node = Node(data)

        if self.head is None:
            self.head = new_Node
            return


        temp = self.head
        while temp.next:
            temp = temp.next
        
        temp.next = new_Node

    def InsertIndex(self, index, data):
        new_Node = Node(data)

        if index < 0:
            print("Invalid Syntax")
            return
        
        if index == 0:
            new_Node.next = self.head
            self.head = new_Node
            return


        temp = self.head

        for i in range(index - 1):

            if temp is None:
                return
            
            temp = temp.next
        
        if temp is None:
            return
        
        new_Node.next = temp.next
        temp.next = new_Node

    def Delete(self,key):
        temp = self.head

        if key == temp:
            temp = temp.next

        while temp.next and temp.next.data != key:
            temp = temp.next

        if temp.next is None:
            print("Invalid Data")
            return
        
        if temp.next.data == key:
            temp.next = temp.next.next
            return
        
    def DeleteIndex(self,index):

        temp = self.head

        if index < 0:
            print("Invalid Syntax")

        if index == 0:
            temp = temp.next

        for i in range(index-1):
            temp = temp.next

        temp.next = temp.next.next

    def Search(self,data):
        temp = self.head
        count = 0


        while temp:
            if temp.data == data:
                print("Data found at index", count)
                return

            count += 1
            temp = temp.next

    

    def Display(self):
        temp = self.head

        while temp:
            print(temp.data,end = " -> ")
            temp = temp.next
        print("Null")

l1 = LinkedList()
l1.InsertEnd(5)
l1.InsertEnd(4)
l1.InsertEnd(3)
l1.InsertEnd(2)
l1.InsertEnd(1)

l1.InsertIndex(0,6)



l1.Display()
